src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -1038,3 +1038,56 @@ const Json::Value& CascadiaSettings::_GetDisabledProfileSourcesJsonObject(const
     }
     return json[JsonKey(DisabledProfileSourcesKey)];
 }
+
+// Method Description:
+// - Write the current state of CascadiaSettings to our settings.json file
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void CascadiaSettings::Export() const
+{
+    const auto json{ ToJson() };
+    _WriteSettings(json.toStyledString());
+}
+
+// Method Description:
+// - Create a new serialized JsonObject from an instance of this class
+// Arguments:
+// - <none>
+// Return Value:
+// the JsonObject representing this instance
+Json::Value CascadiaSettings::ToJson() const
+{
+    // top-level json object
+    // directly inject "globals" and "$schema" into here
+    Json::Value json{ _globals->ToJson() };
+    JsonUtils::SetValueForKey(json, SchemaKey, JsonKey(SchemaValue));
+
+    // "profiles" will always be serialized as an object
+    Json::Value profiles{ Json::ValueType::objectValue };
+    profiles[JsonKey(DefaultSettingsKey)] = _userDefaultProfileSettings->ToJson();
+    Json::Value profilesList{ Json::ValueType::arrayValue };
+    for (const auto& entry : _allProfiles)
+    {
+        const auto prof{ winrt::get_self<implementation::Profile>(entry) };
+        profilesList.append(prof->ToJson());
+    }
+    profiles[JsonKey(ProfilesListKey)] = profilesList;
+    json[JsonKey(ProfilesKey)] = profiles;
+
+    // "schemes" will be an accumulation of _all_ the color schemes

```

Comment:
Just to make sure, "\_all\_ the color schemes" means _including_ Campbell and the other ones in the `defaults.json`, yea? (again, I'm assuming this is something to be sorted out later in "Write settings to JSON/Part 3")

Reply:
Yup. It's gonna be beefy. This'll have to be a part of #8100. I'll link the TODO there for easier tracking.